src/ucp/core/ucp_am.c

```diff
@@ -456,19 +469,57 @@ ucp_am_send_short(ucp_ep_h ep, uint16_t id, uint16_t flags, const void *header,
      * TODO: enable short protocol for such cases when uct_am_short_iov is
      * defined in UCT
      */
-    ucs_assert((length == 0ul) || (header_length == 0));
+    ucs_assert((length == 0ul) || (header_length == 0ul));
     ucs_assert(!(flags & UCP_AM_SEND_REPLY));
-    UCS_STATIC_ASSERT(sizeof(ucp_am_hdr_t) == sizeof(uint64_t));
-    hdr.am_id         = id;
-    hdr.flags         = flags;
-    hdr.header_length = header_length;
+    ucp_am_fill_short_header(&hdr, id, flags, header_length);
 
     sbuf = (header_length != 0) ? (void*)header : (void*)payload;
 
     return uct_ep_am_short(am_ep, UCP_AM_ID_SINGLE, hdr.u64, sbuf,
                            length + header_length);
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_am_send_short_reply(ucp_ep_h ep, uint16_t id, uint16_t flags,
+                        const void *header, size_t header_length,
+                        const void *payload, size_t length)
+{
+    uct_ep_h am_ep = ucp_ep_get_am_uct_ep(ep);
+    size_t tx_length;
+    ucp_am_hdr_t hdr;
+    const void *data;
+    void *tx_buffer;
+    ucs_status_t status;
+
+    ucs_assert(flags & UCP_AM_SEND_REPLY);
+    ucs_assert((length == 0ul) || (header_length == 0ul));
+
+    status = ucp_ep_resolve_remote_id(ep, ep->am_lane);
+    if (ucs_unlikely(status != UCS_OK)) {
+        return status;
+    }
+
+    if (header_length != 0) {
+        tx_length = header_length;
+        data      = header;
+    } else {
+        tx_length = length;
+        data      = payload;
+    }
+
+    tx_buffer = ucs_alloca(tx_length + sizeof(ucs_ptr_map_key_t));
+
+    *((ucs_ptr_map_key_t*)tx_buffer) = ucp_ep_remote_id(ep);
```

Comment:
static assert that ep_id field offset in reply header is equal to the size of short header + add comment to explain this line

Reply:
ok